app/models/category_expert_endorsement.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class CategoryExpertEndorsement < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :endorsed_user, class_name: "User"
+  belongs_to :category
+
+  validate :not_endorsing_self
+
+  after_commit :create_reviewable, on: :create
+
+  private
+
+  def create_reviewable
+    endorsements_for_category = CategoryExpertEndorsement.where(endorsed_user: endorsed_user, category: category).count
+    if endorsements_for_category == SiteSetting.category_expert_suggestion_threshold
```

Comment:
This might not matter depending on the forum it's deployed to, but under load this `==` check might fail, because two endorsements could be created at once and the count would jump by more than one before it's checked here.

If we think that few people will be doing this concurrently we can probably ignore for now.